File: persephone/ip-allocation-service/src/main/java/com/vmware/blockchain/ipam/services/ResourceName.java
/*
 * Copyright (c) 2020 VMware, Inc. All rights reserved. VMware Confidential
 */

package com.vmware.blockchain.ipam.services;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * Data class for resource name.
 */
@AllArgsConstructor
@Data
public class ResourceName {

    private String value = "";
}

File: persephone/ip-allocation-service/src/main/java/com/vmware/blockchain/ipam/server/ResourceName.kt
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc.  All rights reserved. VMware Confidential
 * *************************************************************************/

package com.vmware.blockchain.ipam.server

import kotlinx.serialization.KSerializer
import kotlinx.serialization.Serializable

@Serializable
data class ResourceName (val value: String = "") {
    companion object {
        fun getSerializer(): KSerializer<ResourceName> = serializer()
    }
}